py/pyreplayer.cpp

```diff
@@ -0,0 +1,193 @@
+
+void init_replayer(py::module& m) {
+  py::module m_sub = m.def_submodule("replayer");
+  py::class_<Order>(m_sub, "Order")
+      .def(py::init<>())
+      .def_readwrite("first_frame", &Order::first_frame)
+      .def_readwrite("type", &Order::type)
+      .def_readwrite("targetId", &Order::targetId)
+      .def_readwrite("targetX", &Order::targetX)
+      .def_readwrite("targetY", &Order::targetY)
+      .def(py::self == py::self);
+
+  py::class_<UnitCommand>(m_sub, "UnitCommand")
+      .def(py::init<>())
+      .def_readwrite("frame", &UnitCommand::frame)
+      .def_readwrite("type", &UnitCommand::type)
+      .def_readwrite("targetId", &UnitCommand::targetId)
+      .def_readwrite("targetX", &UnitCommand::targetX)
+      .def_readwrite("targetY", &UnitCommand::targetY)
+      .def_readwrite("extra", &UnitCommand::extra)
+      .def(py::self == py::self);
+
+  py::class_<Unit> unit(m_sub, "Unit");
+  unit.def(py::init<>())
+      .def_readwrite("id", &Unit::id)
+      .def_readwrite("x", &Unit::x)
+      .def_readwrite("y", &Unit::y)
+      .def_readwrite("flags", &Unit::flags)
+      .def_readwrite("health", &Unit::health)
+      .def_readwrite("max_health", &Unit::max_health)
+      .def_readwrite("shield", &Unit::shield)
+      .def_readwrite("max_shield", &Unit::max_shield)
+      .def_readwrite("energy", &Unit::energy)
+      .def_readwrite("maxCD", &Unit::maxCD)
+      .def_readwrite("groundCD", &Unit::groundCD)
+      .def_readwrite("airCD", &Unit::airCD)
+      .def_readwrite("visible", &Unit::visible)
+      .def_readwrite("type", &Unit::type)
+      .def_readwrite("armor", &Unit::armor)
+      .def_readwrite("shieldArmor", &Unit::shieldArmor)
+      .def_readwrite("size", &Unit::size)
+      .def_readwrite("pixel_x", &Unit::pixel_x)
+      .def_readwrite("pixel_y", &Unit::pixel_y)
+      .def_readwrite("pixel_size_x", &Unit::pixel_size_x)
+      .def_readwrite("pixel_size_y", &Unit::pixel_size_y)
+      .def_readwrite("groundATK", &Unit::groundATK)
+      .def_readwrite("airATK", &Unit::airATK)
+      .def_readwrite("groundDmgType", &Unit::groundDmgType)
+      .def_readwrite("airDmgType", &Unit::airDmgType)
+      .def_readwrite("groundRange", &Unit::groundRange)
+      .def_readwrite("airRange", &Unit::airRange)
+      .def_readwrite("orders", &Unit::orders)
+      .def_readwrite("command", &Unit::command)
+      .def_readwrite("velocityX", &Unit::velocityX)
+      .def_readwrite("velocityY", &Unit::velocityY)
+      .def_readwrite("playerId", &Unit::playerId)
+      .def_readwrite("resources", &Unit::resources)
+      .def(
+          "getFlag",
+          [](const Unit& u, Unit::Flags flag) { return (u.flags & flag) > 0; })
+      .def("setFlag", [](Unit& u, Unit::Flags flag, bool value) {
+        if (value)
+          u.flags |= flag;
+        else
+          u.flags &= (~flag);
+      });
+
+  std::string tmp;
+#define DO_FLAG(FLAG)                                                   \
```

Comment:
Yes, but I'd have to do something like do_flag(Units::Flags::Accelerating, "accelerating"), since #FLAG is a string expansion. Would that be cleaner / you prefer that more?